pkg/status/metrics.go

```diff
@@ -40,18 +40,19 @@ func ObserveReleaseConditions(old v1.HelmRelease, new *v1.HelmRelease) {
 		for _, condition := range new.Status.Conditions {
 			conditions[condition.Type] = &condition.Status
 		}
-	}
 
-	for conditionType, conditionStatus := range conditions {
-		labels := stdprometheus.Labels{
-			LabelTargetNamespace: new.GetTargetNamespace(),
-			LabelReleaseName:     new.GetReleaseName(),
-			LabelCondition:       string(conditionType),
-		}
-		if conditionStatus == nil {
-			releaseCondition.Delete(labels)
-		} else {
-			releaseCondition.With(labels).Set(conditionStatusToGaugeValue[*conditionStatus])
+		for conditionType, conditionStatus := range conditions {
```

Comment:
If I'm reading this right, nothing in this function is done if `new` is nil.
Maybe just make a  `if new == nil { return nil }` at the beginning of the function and/or remove the call to `ObserveReleaseConditions` completely from the `DeleteFunc`

Reply:
@hiddeco pointed out that we do want to delete the old release's metrics. Pushed relevant fixes.T